### PR TITLE
MDEV-9350 - 10.0 Fix jemalloc detection for FreeBSD

### PR DESCRIPTION
**Pull request by Sp1l**
As of FreeBSD 10.0 (1000012 https://svnweb.freebsd.org/changeset/base/234924) FreeBSD has jemalloc in base.

### CODE REVIEW

**Comment by svoj** [comment]
Hi Bernard,

Thanks for your contribution. JIRA task has been created to track this pull request: https://mariadb.atlassian.net/browse/MDEV-9350

This task was added to 10.0.24 backlog, which is planned to be handled between 2016-02-10 and 2016-02-17.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license.

The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/

Please indicate in a comment below that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
Sergey

**Comment by Sp1l** [comment]
The contribution of this code and the pull request including one or several files that are new or modified is BSD-new licensed

File: cmake/jemalloc.cmake [hunk 1]
```diff
@@ -20,11 +20,21 @@ MACRO (CHECK_JEMALLOC)
...
       SET(what system)
     ENDIF()
 
-    CHECK_LIBRARY_EXISTS(${libname} malloc_stats_print "" HAVE_JEMALLOC)
+    IF(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND
+      CMAKE_SYSTEM_VERSION STRGREATER "10.0")
+      SET(HAVE_JEMALLOC ON)
+    ELSE()
+      CHECK_LIBRARY_EXISTS(${libname} malloc_stats_print "" HAVE_JEMALLOC)
+    ENDIF()
```
Comment (vuvova) [inline cmake/jemalloc.cmake]: Why is that? What problem did poudriere had with the old code?

Comment (Sp1l) [inline cmake/jemalloc.cmake]: On FreeBSD (as of 10.0) jemalloc is in the base libc so checking malloc_stats_print in a library fails. Earlier patches checked it with libname libc iirc.

**Comment by vuvova** [comment]
done

**Pull request closed without merging**